bin/oneoff/backfill_data/user_cap_state_columns

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+
+# This script migrates CAP data from properties to separate indexed columns.
+
+require 'ruby-progressbar'
+require_relative '../../../dashboard/config/environment'
+
+max_id = User.maximum(:id)
+batch_size = 100_000
+total_updated = 0
+
+progress_bar_title = proc {|num_updates = 0| "Processed[%c/%C]: |%W| Updated: #{total_updated += num_updates} %a"}
+progress_bar = ProgressBar.create(total: max_id, format: progress_bar_title.call)
+
+(1..max_id).step(batch_size) do |start_id|
+  finish_id = [start_id + batch_size.pred, max_id].min
+
+  User.select(:id).where(cap_state: nil).where("properties->>'$.child_account_compliance_state' IS NOT NULL").find_in_batches(start: start_id, finish: finish_id) do |users_batch|
+    User.where(id: users_batch).update_all <<~SQL.strip
```

Comment:
There can be performance tradeoffs when updating records in batches vs individually. Some testing on a clone of the production database cluster can help us assess that. Separately from performance, are we intentionally seeking some of the features of `update_all` such as:

- Do NOT update `created_at` / `updated_at` attributes
- Do NOT invoke callbacks

I think the answer is that we DO want that functionality. These aren't application updates to these records. We're just moving data to a new schema.